app/api/ada/lib/storage/adaMigration.js

```diff
@@ -35,39 +41,63 @@ export async function migrateToLatest(localStorageApi: LocalStorageApi, isImport
     * Therefore mobx may not notice the change as expected
     */
 
-  const migrationMap: { [ver: string]: (() => Promise<void>) } = {
-    '=0.0.1': async () => await testMigration(localStorageApi),
-    '<1.9.0': async () => await moveStorage(localStorageApi, isImported),
-    '<1.4.0': bip44Migration
-  };
+  const migrationMap: Array<[string, () => Promise<boolean>]> = [
+    ['=0.0.1', async () => await testMigration(localStorageApi)],
+    /**
+     * We changed the place where Yoroi data is stored so  we must process this migration first
+     * to ensure other migrations look at the right place for storage
+     */
+    ['<1.9.0', async () => await moveStorage(localStorageApi)],
+    ['<1.4.0', bip44Migration],
+  ];
 
-  for (const key of Object.keys(migrationMap)) {
-    if (semver.satisfies(lastLaunchVersion, key)) {
-      Logger.info(`Migration started for ${key}`);
-      await migrationMap[key]();
+  let appliedMigration = false;
+  for (const entry of migrationMap) {
+    if (satisfies(lastLaunchVersion, entry[0])) {
+      Logger.info(`Migration started for ${entry[0]}`);
+      const applied = await entry[1]();
+      if (applied) {
+        Logger.info(`Applied migration for ${entry[0]}`);
```

Comment:
maybe we should show a message for `failed` (false) too.

Reply:
`false` isn't failed. It's just no action was requoired

Reply:
yeah, but it would be nice to show something. It could be migration `not necessary`. If you see the previous message `Migration started for ...` and then you don’t see anything else, because it wasn’t necessary, it feels that something is missing.